### PR TITLE
Purge unused to create recovery scripts

### PR DESCRIPTION
**Pull request by audrey-denk**

### CODE REVIEW

File: plugins/org.eclipse.osee.jdbc/src/org/eclipse/osee/jdbc/SqlTable.java [hunk 1]
```diff
@@ -203,4 +202,25 @@ private StringBuilder getInsertIntoSqlStart() {
       strB.append(") VALUES (");
       return strB;
    }
+
```
Comment (dondunne) [inline plugins/org.eclipse.osee.jdbc/src/org/eclipse/osee/jdbc/SqlTable.java:205]: Can you add some javadoc explaining what this does?

File: plugins/org.eclipse.osee.orcs.db/src/org/eclipse/osee/orcs/db/internal/health/PurgeUnusedBackingDataAndTransactions.java [hunk 2]
```diff
@@ -156,6 +203,116 @@ public int[] purge() {
...
+      }
+      if (serverPath.equals("null")) {
+         return null;
+      }
+      Path purgeFolder = Paths.get(serverPath + "\\purge");
+      if (Files.exists(purgeFolder)) {
+         serverPath = serverPath + "\\purge";
+      }
+      int i = 0;
+      int[] counts = new int[11];
```
Comment (david-w-miller) [inline plugins/org.eclipse.osee.orcs.db/src/org/eclipse/osee/orcs/db/internal/health/PurgeUnusedBackingDataAndTransactions.java:251]: I think this is a cut/paste size of 11 from above. It looks like it should be 5? Or should there be a list of Strings that it is the sizeOf that could be iterated over?

**Review by david-w-miller — COMMENTED** [review]
Looks good, I had one minor comment in PurgeUnusedBackingDataAndTransactions

File: plugins/org.eclipse.osee.jdbc/src/org/eclipse/osee/jdbc/SqlTable.java (hunk 1 above)
Comment (audrey-denk) [reply plugins/org.eclipse.osee.jdbc/src/org/eclipse/osee/jdbc/SqlTable.java:205]: Done.

File: plugins/org.eclipse.osee.orcs.db/src/org/eclipse/osee/orcs/db/internal/health/PurgeUnusedBackingDataAndTransactions.java (hunk 2 above)
Comment (audrey-denk) [reply plugins/org.eclipse.osee.orcs.db/src/org/eclipse/osee/orcs/db/internal/health/PurgeUnusedBackingDataAndTransactions.java:251]: You are right!  Fixed it : )

**Review by david-w-miller — APPROVED** [review]

File: plugins/org.eclipse.osee.orcs.core/src/org/eclipse/osee/orcs/core/internal/transaction/TransactionFactoryImpl.java [hunk 3]
```diff
@@ -370,75 +408,75 @@ public int compare(ChangeReportRowDto o1, ChangeReportRowDto o2) {
    @Override
    public boolean undoTx(BranchId branch, TransactionId transaction) {
       //@formatter:off
-String query =
```
Comment (lucianovaglienti) [inline plugins/org.eclipse.osee.orcs.core/src/org/eclipse/osee/orcs/core/internal/transaction/TransactionFactoryImpl.java:373]: Can this formatting get removed? So we don't conflict with the mim gamma ids stuff(soon^tm)

File: plugins/org.eclipse.osee.orcs.rest.model/src/org/eclipse/osee/orcs/rest/model/RelationEndpoint.java [hunk 4]
```diff
@@ -91,5 +91,10 @@ Response createRelationByType(@PathParam("sideA") ArtifactId sideA, @PathParam("
    List<RelationTypeToken> convertRelations(@PathParam("artifactA") ArtifactId artifactA,
       @PathParam("oldRelationType") RelationTypeToken oldRelationType,
       @PathParam("newRelationType") RelationTypeToken newRelationType);
+   
+ @GET
```
Comment (lucianovaglienti) [inline plugins/org.eclipse.osee.orcs.rest.model/src/org/eclipse/osee/orcs/rest/model/RelationEndpoint.java:95]: This is kinda weird spot to put this...

File: plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/TransactionEndpointImpl.java [hunk 5]
```diff
@@ -155,8 +152,8 @@ public Response purgeTxs(String txIds) {
    @Override
    public Response purgeUnusedBackingDataAndTransactions() {
       orcsApi.userService().requireRole(CoreUserGroups.OseeAccessAdmin);
-      orcsApi.getTransactionFactory().purgeUnusedBackingDataAndTransactions();
-      return Response.ok().build();
+      int[] rtn = orcsApi.getTransactionFactory().purgeUnusedBackingDataAndTransactions();
```
Comment (lucianovaglienti) [inline plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/TransactionEndpointImpl.java:155]: Can we inline this or handle potential errors here?

Comment (audrey-denk) [reply plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/TransactionEndpointImpl.java:155]: Previously this was returning this kind of array which is somewhat useful.  Perhaps I could update it to return a pojo with more details about what was purged when I get around to updating to purge unused transactions.

File: plugins/org.eclipse.osee.orcs.rest.model/src/org/eclipse/osee/orcs/rest/model/RelationEndpoint.java (hunk 4 above)
Comment (audrey-denk) [reply plugins/org.eclipse.osee.orcs.rest.model/src/org/eclipse/osee/orcs/rest/model/RelationEndpoint.java:95]: With the removal of the DB Health Endpoint, this seemed the best spot.  It isn't currently used, so when we decide to start using it, we can move it if needed.

File: plugins/org.eclipse.osee.orcs.core/src/org/eclipse/osee/orcs/core/internal/transaction/TransactionFactoryImpl.java (hunk 3 above)
Comment (lucianovaglienti) [reply plugins/org.eclipse.osee.orcs.core/src/org/eclipse/osee/orcs/core/internal/transaction/TransactionFactoryImpl.java:373]: No, what I'm asking is, please don't format this file, as you are going to cause merge conflicts on the mim gamma id branch where we fixed a bunch of issues related to transactions.

Comment (audrey-denk) [reply plugins/org.eclipse.osee.orcs.core/src/org/eclipse/osee/orcs/core/internal/transaction/TransactionFactoryImpl.java:373]: oh gotcha... ok updated it.

**Review by dondunne — APPROVED** [review]

**Review by lucianovaglienti — APPROVED** [review]

**Pull request merged**